apps/studio/src/server/modules/user/inactiveUsers.service.ts

```diff
@@ -72,127 +70,172 @@ export const getInactiveUsers = async ({
     .execute()
 }
 
-interface DeactivateUserProps {
+interface DeactivateUsersProps {
+  tx: Transaction<DB>
+  userIds: User["id"][]
+}
+type DeactivateUsersOutput = {
   user: User
-  userIdsToDeactivate: User["id"][] // used for idempotency purposes
+  siteIds: Site["id"][]
+}[]
+
+export const deactivateUsers = async ({
+  tx,
+  userIds,
+}: DeactivateUsersProps): Promise<DeactivateUsersOutput> => {
+  // prevent empty array from being passed in
+  if (userIds.length === 0) return []
+
+  const deletedPermissions = await tx
+    .updateTable("ResourcePermission")
+    .where("userId", "in", userIds)
+    .where("deletedAt", "is", null)
+    .set({ deletedAt: new Date() })
+    .returningAll()
+    .execute()
+
+  return tx
+    .selectFrom("User")
+    .where("id", "in", userIds)
+    .selectAll()
+    .execute()
+    .then((users) =>
+      users.map((user) => ({
+        user,
+        siteIds: deletedPermissions
+          .filter((permission) => permission.userId === user.id)
+          .map((permission) => permission.siteId),
+      })),
+    )
 }
```

Comment:
[nit] just for consistency's sake, can we avoid mixing `await` and `then` syntax? 

```ts
const users = tx.updateTable(...).execute()
return users.map(...)
```

Reply:
updated in https://github.com/opengovsg/isomer/pull/1489/commits/6c8ef4c16b06d64aaece7914c7764ac4d8276f35